Automattic/wp-calypso
# Improve keyboard support for the language switcher

#### Changes proposed in this Pull Request

This PR improves keyboard support within the language switcher modal.

The main improvements are:

-   Expand search field when start typing
-   Select language if its slug matches search   
-   Confirm selection with Enter key
-   Navigating through search results with keyboard arrow keys

#### Screenshots

Before: 
![Screen Capture on 2020-01-06 at 14-24-36](https://user-images.githubusercontent.com/2722412/71818542-fd858a80-3091-11ea-962c-074ca82ec69e.gif)

After:
![Screen Capture on 2020-01-10 at 12-16-23](https://user-images.githubusercontent.com/2722412/72145946-2fedfb00-33a4-11ea-8e98-449485df8ca7.gif)


#### Testing instructions

-   Check if tests pass
-   Open http://calypso.localhost:3000/
-   Go to Profile → Account Settings → Interface Language
-   Confirm that UI behaves as described

#### TODO

- [x] Expand search field when start typing
- [x] Select language if its slug matches search
- [x] Confirm selection with Enter key
- [x] Navigating through search results with keyboard arrow keys
